app/src/main/java/app/security/RequiresPermissionsAnnotationRule.java

```diff
@@ -63,8 +63,21 @@ public Publisher<SecurityRuleResult> check(HttpRequest<?> request,
         List<String> declaredPermissionsAsStrings = declaredPermissions.stream()
                 .map(Enum::toString).collect(Collectors.toList());
 
-        if (unityAuthService.isUserPermittedForAction(bearerToken, resolveJurisdictionId(request),
-                declaredPermissionsAsStrings)) {
+        String jurisdictionId = request.getParameters().get("jurisdiction_id");
+        String tenantId = request.getParameters().get("tenant_id");
+        if (jurisdictionId == null && tenantId == null) {
+            throw new IllegalArgumentException(
+                    "The Jurisdiction Id must exists as a query parameter when using @RequiresPermissions");
```

Comment:
Should be "Either the Jurisdiction Id or Tenant Id must exist as a query parameter when using @RequiresPermissions"